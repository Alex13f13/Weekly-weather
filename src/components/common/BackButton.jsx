import KeyboardBackspaceIcon from "@mui/icons-material/KeyboardBackspace";
import IconButton from "@mui/material/IconButton";
import { useNavigate } from "react-router-dom";

export default function BackButton() {
	const navigate = useNavigate();
	const handleGoBack = () => {
		navigate(-1);
	};

	return (
		<IconButton onClick={handleGoBack} aria-label="back" size="large">
			<KeyboardBackspaceIcon />
		</IconButton>
	);
}
