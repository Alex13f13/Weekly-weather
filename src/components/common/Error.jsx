import styled from "styled-components";

const ErrorNoDataContainer = styled.div`
	width: 90%;
	height: auto;
	background-color: #f8d7da;
	color: #721c24;
	padding: 10px;
	border: 1px solid #f5c6cb;
	border-radius: 4px;
	margin-top: 20px;
`;

const ErrorMessage = styled.p`
	margin: 0;
	text-align: center;
`;

export default function ErrorNoData() {
	return (
		<ErrorNoDataContainer>
			<ErrorMessage>Hubo un error al cargar los datos.</ErrorMessage>
		</ErrorNoDataContainer>
	);
}
