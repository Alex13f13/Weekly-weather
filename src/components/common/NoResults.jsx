export default function NoResults() {
	return <p>No hay resultados</p>;
}
