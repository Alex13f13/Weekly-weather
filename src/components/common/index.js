export { default as BackButton } from './BackButton';
export { default as DataDisplay } from './DataDisplay';
export { default as Error } from './Error';
export { default as FavButton } from './FavButton';
export { default as Loading } from './Loading';
export { default as NoResults } from './NoResults';
export { default as Selector } from './Selector';
export { default as StatusManager } from './StatusManager';
export { default as WeatherCard } from './WeatherCard';
export { default as WeatherCardList } from './WeatherCardList';
