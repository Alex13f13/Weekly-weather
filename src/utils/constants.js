export const HTML_TAGS = {
	p: "p",
	span: "span",
	div: "div",
	h1: "h1",
	h2: "h2",
	h3: "h3",
	h4: "h4",
	h5: "h5",
	h6: "h6",
};

export const SIZE = {
  small: "small",
  medium: "medium",
  large: "large",
};

export const LABEL = {
	back: "back",
	favourite: "favourite",
	max: "MAX:",
	min: "MIN:",
	province: "Provincia",
	temperature: "Temperatura:",
	temperatureMax: "Temp. M치xima:",
	temperatureMin: "Temp. M칤nima:",
	humidity: "Humedad:",
	uv: "칈ndice max. uv:",
	wind: "Viento:",
	loading: "Cargando...",
	error: "Hubo un error al cargar los datos",
	noResults: "No hay resultados",
	seletProvince: "Selecciona una provincia 游깴",
	favListButton: "Lista de favoritos",
	nextDays: "Pr칩ximos d칤as",
  seeMore: "Ver m치s",
};

export const UNITS = {
  celsius: "춿C",
  percent: "%",
  kmh: "km/h",
};

export const VARIANT = {
	primary: "primary",
	secondary: "secondary",
	success: "success",
	danger: "danger",
	warning: "warning",
	info: "info",
	standard: "standard",
  outlined: "outlined",
};

export const PRIVINCE = {
  15: "A Coru침a",
  3: "Albacete",
  4: "Alicante",
  1: "Almer칤a",
  33: "Asturias",
  5: "츼vila",
  6: "Badajoz",
  7: "Baleares",
  8: "Barcelona",
  9: "Burgos",
  10: "C치ceres",
  11: "C치diz",
  39: "Cantabria",
  12: "Castell칩n",
  51: "Ceuta",
  13: "Ciudad Real",
  14: "C칩rdoba",
  16: "Cuenca",
  20: "Gipuzkoa",
  17: "Girona",
  18: "Granada",
  19: "Guadalajara",
  21: "Huelva",
  22: "Huesca",
  23: "Ja칠n",
  26: "La Rioja",
  35: "Las Palmas",
  24: "Le칩n",
  25: "Lleida",
  27: "Lugo",
  28: "Madrid",
  29: "M치laga",
  52: "Melilla",
  30: "Murcia",
  31: "Navarra",
  32: "Ourense",
  34: "Palencia",
  36: "Pontevedra",
  37: "Salamanca",
  38: "Santa Cruz de Tenerife",
  40: "Segovia",
  41: "Sevilla",
  42: "Soria",
  43: "Tarragona",
  44: "Teruel",
  45: "Toledo",
  46: "Valencia",
  47: "Valladolid",
  48: "Vizcaya",
  49: "Zamora",
  50: "Zaragoza",
};

