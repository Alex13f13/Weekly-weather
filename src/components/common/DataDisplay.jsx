import { createElement } from "react";
import { HTML_TAGS } from "../../utils/constants";

export default function DataDisplay({ data, label = "", units = "", tag = HTML_TAGS.p }) {
	const selectedTag = HTML_TAGS[tag] || HTML_TAGS.p;

	return data && createElement(selectedTag, null, `${label} ${data} ${units}`);
}
