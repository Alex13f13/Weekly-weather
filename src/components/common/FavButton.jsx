import IconButton from "@mui/material/IconButton";
import FavoriteIcon from "@mui/icons-material/Favorite";
import HeartBrokenIcon from "@mui/icons-material/HeartBroken";
import { SIZE, LABEL } from "../../utils/constants";

export default function FavButton({ isFavorite, onClick = () => {} }) {
	return (
		<IconButton aria-label={LABEL.favourite} size={SIZE.large} onClick={onClick}>
			{isFavorite ? <HeartBrokenIcon /> : <FavoriteIcon />}
		</IconButton>
	);
}
