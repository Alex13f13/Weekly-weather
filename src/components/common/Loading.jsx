import { LABEL } from "../../utils/constants";

export default function Loading() {
	return <>{LABEL.loading}</>;
}
