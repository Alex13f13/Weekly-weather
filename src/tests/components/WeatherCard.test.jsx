import { render, screen, fireEvent } from "@testing-library/react";
import { WeatherCard } from "../../components/common";
import { LABEL, UNITS } from "../../utils/constants";
import { vi } from "vitest";

describe("WeatherCard Component", () => {
	const defaultProps = {
		icon: <div>Icon</div>,
		weather: "Sunny",
		max: "30",
		min: "20",
		name: "City Name",
	};

	const renderWeatherCard = (props = {}) => {
		const combinedProps = { ...defaultProps, ...props };
		return render(<WeatherCard {...combinedProps} />);
	};

	it("renders WeatherCard component with the correct content", () => {
		renderWeatherCard();

		const card = screen.getByTestId("weather-card");
		const cityName = screen.getByText(defaultProps.name);
		const weatherText = screen.getByText(defaultProps.weather);
		const maxTemp = screen.getByText(`${LABEL.max} ${defaultProps.max} ${UNITS.celsius}`);
		const minTemp = screen.getByText(`${LABEL.min} ${defaultProps.min} ${UNITS.celsius}`);

		expect(card).toBeDefined();
		expect(cityName).toBeDefined();
		expect(weatherText).toBeDefined();
		expect(maxTemp).toBeDefined();
		expect(minTemp).toBeDefined();
	});

	it("calls the onClick function when the card is clicked", () => {
		const onClickMock = vi.fn();

		renderWeatherCard({ onClick: onClickMock });

		const card = screen.getByTestId("weather-card");

		fireEvent.click(card);

		expect(onClickMock).toHaveBeenCalled();
	});

	it("renders WeatherCard component if props are missing", () => {
		renderWeatherCard({
			icon: undefined,
			weather: undefined,
			max: undefined,
			min: undefined,
			name: undefined,
		});

		const card = screen.getByTestId("weather-card");
		const cityName = screen.queryByText(defaultProps.name);
		const weatherText = screen.queryByText(defaultProps.weather);
		const maxTemp = screen.queryByText(`${LABEL.max} ${defaultProps.max} ${UNITS.celsius}`);
		const minTemp = screen.queryByText(`${LABEL.min} ${defaultProps.min} ${UNITS.celsius}`);

		expect(card).toBeDefined();
		expect(cityName).toBeNull();
		expect(weatherText).toBeNull();
		expect(maxTemp).toBeNull();
		expect(minTemp).toBeNull();
	});

	it("matches snapshot", () => {
		const { container } = renderWeatherCard();
		expect(container).toMatchSnapshot();
	});
});
