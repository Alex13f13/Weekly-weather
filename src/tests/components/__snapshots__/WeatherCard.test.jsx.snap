// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WeatherCard Component > matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-testid="weather-card"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="sc-dkjKgF hzxYih"
      >
        <h4>
           City Name 
        </h4>
      </div>
      <div
        class="sc-hLclGa eqCwFX"
      >
        <div>
          Icon
        </div>
        <h5>
           Sunny 
        </h5>
      </div>
      <hr />
      <div
        class="sc-eDnVMP jKGaQK"
      >
        <p>
          MAX: 30 °C
        </p>
        <p>
          MIN: 20 °C
        </p>
      </div>
    </div>
  </div>
</div>
`;
